app/controllers/admin/works_controller.rb

```diff
@@ -119,7 +121,48 @@ def download_ohms_xml
     send_data @work.oral_history_content!.ohms_xml_text,
       :type => 'text/xml; charset=UTF-8;',
       :disposition => ContentDisposition.format(disposition: "attachment", filename: "#{@work.oral_history_content!.ohms_xml.accession}.xml")
-      #:disposition => "attachment; filename=#{@work.oral_history_content!.ohms_xml.accession}.xml"
+  end
+
+  # PATCH/PUT /admin/works/ab2323ac/submit_ohms_xml
+  def submit_searchable_transcript_source
+    unless params[:searchable_transcript_source].present?
+      redirect_to admin_work_path(@work, anchor: "nav-oral-histories"), flash: { error: "No file received" }
+      return
+    end
+    transcript = params[:searchable_transcript_source].read
+    # Very basic check for now. Not stripping HMTL tags,
+    # even though they really don't belong in these transcripts.
+    #transcript_checks = [true]
+    transcript_checks = [
+      (params[:searchable_transcript_source].content_type.start_with?('text/')),
+      ((MimeMagic.by_magic(transcript).nil?)||(MimeMagic.by_magic(transcript).to_s.start_with? 'text/')),
```

Comment:
this is going overboard I think, just trust the content_type, no need to bring mime_magic into it. Which should simplify this code a lot. Let's keep it simple, less code to maintain, less code that can have bugs! no need for `transcript_checks`, just check the content type.

Reply:
No problem -- removed.